### ISSUE TITLE
The memory is not released after the process ends？

### ISSUE THREAD

**Issue by chushuai** [pyppeteer/pyppeteer#278]
When I use it to take screenshots of a large number of websites, 
the memory is still not released after all the processes are finished. 
The following figure shows the memory usage of the machine

![image](https://user-images.githubusercontent.com/5875458/123052776-c8135f00-d435-11eb-8cdb-7524c3796ef3.png)

**Comment by alenpaulvarghese** [comment]
please attach your code.

**Comment by chushuai** [comment]
Here's my code @alenpaul2001 

import pyppeteer
from pyppeteer import launch


class Snapshot(object):

    def __init__(self):
        self.result_list = []
        self.contex = None

    async def get_url(self, target):
        try:
            page = await self.contex.newPage()
            await page.setCacheEnabled(False)
            await page.setUserAgent(
                "Mozilla/5.0 (Windows NT 6.0) AppleWebKit/536.5 (KHTML, like Gecko) Chrome/18.0.1084.36 Safari/536.5")
            await page.setViewport(viewport={'width': 1536, 'height': 768})
            url = target["url"]
            async with timeout(30):
                await page.goto(url)
                result = await page.screenshot(type="png")
                target["image"] = result
                self.result_list.append(target)
            page.close()
        except pyppeteer.errors.PageError as e:
            print(traceback.format_exc())
        except asyncio.TimeoutError as e:
            print(traceback.format_exc())
        except:
            print(traceback.format_exc())
        return None

    def kill_proc_tree(self, pid):
        try:
            root_proc = psutil.Process(pid)
            procs = root_proc.children(recursive=True)
            for p in procs:
                p.send_signal(psutil.signal.SIGKILL)
        except Exception:
            print(traceback.format_exc())

    async def run(self, target_list):
        try:
            while True:
                try:
                    self.browser = await launch(args=['--no-sandbox', '--window-size=1920,1080'],
                                                ignoreHTTPSErrors=True, headless=True)
                    self.contex = await  self.browser.createIncognitoBrowserContext()
                except:
                    # print(traceback.format_exc())
                    continue
                break

            tasks = [self.get_url(target) for target in target_list]
            await asyncio.gather(*tasks)
        except:
            # print(traceback.format_exc())
            pass
        self.kill_proc_tree(os.getpid())
        return None

**Comment by alenpaulvarghese** [comment]
clear result_list?

**Comment by chushuai** [comment]
This process killed, but memory is not free @alenpaul2001

**Comment by alenpaulvarghese** [comment]
Are you sure python is using the memory?
there is no trace in ur screenshot?

**Comment by chushuai** [comment]
I think it's the browser cache

**Comment by alenpaulvarghese** [comment]
@chushuai  I don't think so, the browser is defaulted to incognito mode so it won't save any cache. please double check your design choice.

**Comment by chushuai** [comment]
What bothers me is that all the processes involved are dead, so why can't the memory drop
